File: py/二分查找.py
# Author: Chenchen

# 二分法查找
def bin_search(data_li, value):
  low = 0
  high = len(data_li) - 1
  while low <= high:
    mid = (low + high) // 2
    if data_li[mid] == value:
      return mid
    elif data_li[mid] >value:
      high = mid-1
    else:
      low = mid+1
  return None

# print(bin_search(list(range(0,10)),5))

File: js/二分查找.js
// 二分法查找

const binSearch = function (data_set,val){
  let low = 0;
  let high = data_set.length;
  while(low<=high){
    let mid = Math.floor((low + high)/2)
    if(data_set[mid] === val){
      return mid
    }else if(data_set[mid] > val){
      high = mid -1;
    }else{
      low = mid +1
    }
  }
  return null
}

// console.log(bin_search([1,2,3,4,5,6,7,8,9],1))

